python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -337,7 +343,13 @@ def dist_inference_one_layer(layer_id, g, dataloader, target_ntypes, layer, get_
             if k in output_nodes:
                 assert k in y, "All mini-batch outputs should have the same tensor names."
                 y[k][output_nodes[k]] = h[k].cpu()
-    flush_data()
+    # The nodes are split in such a way that all processes only need to compute
+    # the embeddings of the nodes in the local partition. Therefore, a barrier
+    # is enough to ensure that all data have been written to memory for distributed
+    # read after this function is returned.
+    # Note: there is a risk here. If the nodes for inference on each partition
+    # are very skewed, some of the processes may timeout in the barrier.
+    barrier()
```

Comment:
You assume the data are split with `force_even=False`.
If a custom inference code invoke this API with force_even=True will potentially trigger a bug. Are we going to force users to use `force_even=False`? Is there anyway to check it?

Reply:
this is a problem. we don't have a way to effectively check if the data is split evenly or not.